core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -1238,7 +1248,90 @@ protected void postProcessing(boolean closeEarly) {
             loadOrPrepareCH(closeEarly);
     }
 
+
     protected void importPublicTransit() {
+
+        //We use that method to import speeds
+        if(speedsProvider != null){
+            logger.info("Start Custom Speeds Provider");
+            long startTime = System.nanoTime();
+
+            List<String> vehicles = this.getProfiles().stream().map(Profile::getVehicle).distinct().collect(Collectors.toList());
```

Comment:
Fail if profiles is empty?

Reply:
Not sure, if profiles is empty GH will fail on init before calling this internal method. It's mandatory to have at least one profile to build a graph